app/scripts/background.js

```diff
@@ -918,6 +980,31 @@ async function onInstall() {
     addAppInstalledEvent();
     platform.openExtensionInBrowser();
   }
+  onNavigateToTab();
+}
+
+function onNavigateToTab() {
+  browser.tabs.onActivated.addListener((onActivatedTab) => {
+    if (controller) {
+      const { tabId } = onActivatedTab;
+      const currentOrigin = tabOriginMapping[tabId];
+      // *** Emit DappViewed metric event when ***
+      // - navigate to a connected dapp
+      if (currentOrigin) {
+        const connectSitePermissions =
+          controller.permissionController.state.subjects[currentOrigin];
+        // when the dapp is not connected, connectSitePermissions is undefined
+        const isConnectedToDapp = connectSitePermissions !== undefined;
+        if (isConnectedToDapp) {
```

Comment:
does the "New Tab" check need to happen here too?

Reply:
not really for this one. I added a comment above, this is only triggered when we navigate to a connected dapp, which means you had this tab open already, and just visit it again ~